src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -679,6 +682,10 @@ internal ExitWriteLock DrainRunningQueries()
 
                 OnInitialization();
 
+                LastDatabaseEtagOnIndexCreation = _environment.IsNew 
+                    ? LastDatabaseEtag.Save(DocumentDatabase, this) 
```

Comment:
I don't particularly like the `Save` here. You are are mutating data in a way that is very subtle and hard to see. 
I would rather create this explicitly.

Reply:
Done